apps/ledger-live-desktop/src/renderer/families/elrond/StakeBanner.tsx

```diff
@@ -0,0 +1,92 @@
+
+import { getAccountBannerState as getElrondBannerState } from "@ledgerhq/live-common/families/elrond/banner";
+import { AccountBanner } from "~/renderer/screens/account/AccountBanner";
+import React from "react";
+import { StakeAccountBannerParams } from "~/renderer/screens/account/types";
+import { useElrondPreloadData } from "@ledgerhq/live-common/families/elrond/react";
+import { ElrondAccount } from "@ledgerhq/live-common/families/elrond/types";
+import { track } from "~/renderer/analytics/segment";
+import { stakeDefaultTrack } from "~/renderer/screens/stake/constants";
+import { useDispatch } from "react-redux";
+import { openModal } from "~/renderer/actions/modals";
+import { modals } from "~/renderer/families/elrond/modals";
+import { AccountBannerState } from "@ledgerhq/live-common/lib/families/elrond/banner";
+
+export const StakeBanner: React.FC<{ account: ElrondAccount }> = ({ account }) => {
+  const { t } = useTranslation();
+  const stakeAccountBanner = useFeature("stakeAccountBanner");
+  const stakeAccountBannerParams: StakeAccountBannerParams | null =
+    stakeAccountBanner?.params ?? null;
+
+  const elrondPreloadData = useElrondPreloadData();
+  const dispatch = useDispatch();
+  const bannerState: AccountBannerState = getElrondBannerState(account, elrondPreloadData);
+  const showDelegationBanner = bannerState.bannerType === "delegate";
+  if (!stakeAccountBannerParams?.elrond?.delegate && showDelegationBanner) {
+    return null;
+  }
+
+  if (!stakeAccountBannerParams?.elrond?.redelegate && bannerState.bannerType === "redelegate") {
+    return null;
+  }
+
+  const title = showDelegationBanner
+    ? t("account.banner.delegation.title")
+    : t("account.banner.redelegation.elrond.title");
+  const description = showDelegationBanner
+    ? t("account.banner.delegation.elrond.description")
+    : t("account.banner.redelegation.elrond.description");
+  const cta = showDelegationBanner
+    ? t("account.banner.delegation.cta")
+    : t("account.banner.redelegation.cta");
+  const linkUrl = showDelegationBanner
+    ? "https://www.ledger.com/staking"
+    : "https://support.ledger.com/hc/en-us/articles/7228337345693-Staking-MultiversX-EGLD-through-Ledger-Live-?support=true";
+  const linkText = showDelegationBanner
+    ? t("account.banner.delegation.linkText")
+    : t("account.banner.redelegation.linkText");
+  const onClick = () => {
+    track("button_clicked", {
+      ...stakeDefaultTrack,
+      delegation: "stake",
+      page: "Page Account",
+      button: "delegate",
+      redelegate: bannerState.bannerType === "redelegate",
+      token: "ELROND",
+    });
+
+    if (bannerState.bannerType === "redelegate") {
+      dispatch(
+        openModal(modals.unstake, {
+          account,
+          contract: bannerState.selectedDelegation.contract,
+          validators: elrondPreloadData.validators,
+          delegations: bannerState.mappedDelegations,
```

Comment:
do we need to pass ```mappedDelegations``` from state? Is it not an "account" attribute?

Reply:
It is an account attribute that has been transformed. Annoyingly the modal doesn't open properly otherwise. This is duplicating this logic here:

https://github.com/LedgerHQ/ledger-live/blob/develop/apps/ledger-live-desktop/src/renderer/families/elrond/EarnRewards.jsx?plain=1#L86-L89

Reply:
Should we "obtain" "mappedDelegations" directly in StakeBanner.tsx as it is independent of the banner state?

Reply:
Yes we can do. I had it that way originally but abstracted to the helper - but I actually think it's better in StakeBanner as it seems like we need this just for LLD to work properly.

Reply:
https://github.com/LedgerHQ/ledger-live/pull/2672/commits/e2109f0e00214e2c5465f6008f16f9c71c3e0c5f